app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id'
+  has_many :comments
+
+  validates :author_id, presence: true
+  validates :title, presence: true
+  validates :text, presence: true
+  # A method that updates the posts counter for a user.
+  def update_posts_counter
+    update(posts_counter: posts.count)
+  end
```

Comment:
- [OPTIONAL] It would be nice if you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new comment has been saved to database, then make the `update_post_counter` method private.
Something like this :point_down: 

```
    after_save :update_post_counter

    private

    def update_post_counter
        author.increment!(:posts_counter)
    end
```